90zfsbootmenu/zfsbootmenu-parse-commandline.sh

```diff
@@ -21,10 +21,31 @@ fi
 
 # Let the command line override our host id.
 # shellcheck disable=SC2034
-spl_hostid=$(getarg spl_hostid=)
+cli_spl_hostid=$( getarg spl.spl_hostid ) || cli_spl_hostid=$( getarg spl_hostid )
+if [ -n "${cli_spl_hostid}" ] ; then
+  # Tests modify the value, so preserve this for a warning
+  orig_cli_spl_hostid="${cli_spl_hostid}"
+
+  # Start empty so only valid hostids are set for future use
+  spl_hostid=
+
+  # Test for decimal
+  if (( 10#${cli_spl_hostid} )) >/dev/null 2>&1 ; then
+    spl_hostid="$( printf "%08x" "${cli_spl_hostid}" )"
+  # Test for hex. Requires 0x, if present, to be stripped
+  # The change to cli_spl_hostid isn't saved outside of the test
+  elif (( 16#${cli_spl_hostid//0x/} )) >/dev/null 2>&1 ; then
```

Comment:
1. Probably better just to do `${cli_spl_hostid#0x}` since accepting something like `0x0xdead0xbeef` as `deadbeef` seems dubious.
2. `(( 10#0 ))` and `(( 16#0 ))` both silently return `1` and cause the tests to fail. What's the right response here? Setting `spl_hostid=0` might indicate a desire to force all zeros in `/etc/hostid`.

Reply:
Good catch, I'll special-case 0 / 0x0. I can change the `//0x/` to just remove a single prefix. It's fairly unlikely that we'd ever need to protect against a random smattering of `0x` spread throughout the hostid string.